File: src/main/java/com/regionrental/commands/ExtendCommand.java
package com.regionrental.commands;

import com.regionrental.RegionRental;
import com.regionrental.managers.Rental;
import net.milkbowl.vault.economy.Economy;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class ExtendCommand implements CommandExecutor {
    
    private final RegionRental plugin;
    
    public ExtendCommand(RegionRental plugin) {
        this.plugin = plugin;
    }
    
    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (!(sender instanceof Player)) {
            sender.sendMessage(ChatColor.RED + "This command can only be used by players!");
            return true;
        }
        
        Player player = (Player) sender;
        
        if (!player.hasPermission("regionrental.extend")) {
            player.sendMessage(plugin.getConfigManager().getMessage("no-permission"));
            return true;
        }
        
        if (args.length < 1) {
            player.sendMessage(ChatColor.RED + "Usage: /rrextend <region>");
            return true;
        }
        
        String regionName = args[0];
        Rental rental = plugin.getRentalManager().getRental(regionName);
        
        if (rental == null) {
            player.sendMessage(ChatColor.RED + "Region " + regionName + " is not rented!");
            return true;
        }
        
        // Check if player owns this rental
        if (!rental.getPlayerUUID().equals(player.getUniqueId())) {
            player.sendMessage(ChatColor.RED + "You don't own this rental!");
            return true;
        }
        
        // Check extension limit
        if (rental.getExtensionCount() >= plugin.getConfigManager().getMaxExtensions()) {
            player.sendMessage(plugin.getConfigManager().getMessage("max-extensions-reached"));
            return true;
        }
        
        // Get extension price
        double price = plugin.getConfigManager().getPriceForRegion(regionName);
        double multiplier = plugin.getConfig().getDouble("extension.price-multiplier", 1.0);
        price = price * multiplier;
        
        // Check economy
        Economy economy = plugin.getEconomy();
        if (economy == null) {
            player.sendMessage(ChatColor.RED + "Economy system not available!");
            return true;
        }
        
        if (!economy.has(player, price)) {
            player.sendMessage(plugin.getConfigManager().getMessage("not-enough-money",
                "{amount}", String.format(plugin.getConfigManager().getCurrencyFormat(), price)));
            return true;
        }
        
        // Withdraw money
        economy.withdrawPlayer(player, price);
        
        // Extend rental
        int days = plugin.getConfigManager().getExtensionDuration();
        if (plugin.getRentalManager().extendRental(regionName, player, days, price)) {
            player.sendMessage(plugin.getConfigManager().getMessage("rental-extended",
                "{region}", regionName,
                "{days}", String.valueOf(days),
                "{price}", String.format(plugin.getConfigManager().getCurrencyFormat(), price)));
        } else {
            // Refund if extension failed
            economy.depositPlayer(player, price);
            player.sendMessage(ChatColor.RED + "Failed to extend rental!");
        }
        
        return true;
    }
}

File: src/main/kotlin/com/zonerental/commands/ExtendCommand.kt
package com.zonerental.commands

import com.zonerental.ZoneRental
import com.zonerental.extensions.asPlayerOrNull
import com.zonerental.extensions.sendMiniMessage
import com.zonerental.util.WorldRegionParser
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

class ExtendCommand(private val plugin: ZoneRental) : CommandExecutor {

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        val player = sender.asPlayerOrNull() ?: run {
            sender.sendMiniMessage("<red>This command can only be used by players!")
            return true
        }

        if (!player.hasPermission("zonerental.extend")) {
            player.sendMessage(plugin.configManager.getMessage("no-permission"))
            return true
        }

        if (args.isEmpty()) {
            player.sendMiniMessage("<red>Usage: /zrextend <region>")
            player.sendMiniMessage("<yellow>Example: /zrextend shop1")
            return true
        }

        // Parse region argument with world inference (player's world)
        val parsed = WorldRegionParser.parse(args[0], player) ?: run {
            player.sendMiniMessage("<red>Invalid region format!")
            return true
        }

        val world = parsed.getWorld() ?: run {
            player.sendMiniMessage("<red>World not found!")
            return true
        }
        val regionName = parsed.regionName

        val rental = plugin.rentalManager.getRental(regionName, world)

        if (rental == null) {
            player.sendMiniMessage("<red>Region $regionName is not rented!")
            return true
        }

        // Check if player owns this rental
        if (rental.playerUUID != player.uniqueId) {
            player.sendMiniMessage("<red>You don't own this rental!")
            return true
        }

        // Check extension limit
        if (rental.extensionCount >= plugin.configManager.maxExtensions) {
            player.sendMessage(plugin.configManager.getMessage("max-extensions-reached"))
            return true
        }

        // Get extension price
        val basePrice = plugin.configManager.getPriceForRegion(regionName, world)
        val multiplier = plugin.config.getDouble("extension.price-multiplier", 1.0)
        val price = basePrice * multiplier

        // Check economy
        val economy = plugin.economy
        if (economy == null) {
            player.sendMiniMessage("<red>Economy system not available!")
            return true
        }

        if (!economy.has(player, price)) {
            val formattedPrice = String.format(plugin.configManager.currencyFormat, price)
            player.sendMessage(plugin.configManager.getMessage("not-enough-money", "{amount}", formattedPrice))
            return true
        }

        // Withdraw money
        economy.withdrawPlayer(player, price)

        // Extend rental
        val days = plugin.configManager.extensionDuration
        if (plugin.rentalManager.extendRental(regionName, world, player, days, price)) {
            val formattedPrice = String.format(plugin.configManager.currencyFormat, price)
            player.sendMessage(plugin.configManager.getMessage("rental-extended",
                "{region}", regionName,
                "{days}", days.toString(),
                "{price}", formattedPrice))
        } else {
            // Refund if extension failed
            economy.depositPlayer(player, price)
            player.sendMiniMessage("<red>Failed to extend rental!")
        }

        return true
    }
}
